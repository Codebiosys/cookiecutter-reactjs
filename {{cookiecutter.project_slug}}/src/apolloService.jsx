import {
  ApolloClient,
  createNetworkInterface,
} from "react-apollo";

const GRAPHQL_HOST = process.env.GRAPHQL_HOST || window.location.hostname;
const GRAPHQL_PORT = process.env.GRAPHQL_PORT || 4000;
const GRAPHQL_ENDPOINT = process.env.GRAPHQL_ENDPOINT|| "/graphql";

// GRAPHQL_HOST & GRAPHQL_PORT are defined in webpack
const networkInterface = createNetworkInterface({
  uri: `${GRAPHQL_HOST}:${GRAPHQL_PORT}${GRAPHQL_ENDPOINT}`
});

export default new ApolloClient({
  networkInterface,
  queryDeduplication: true,
});
