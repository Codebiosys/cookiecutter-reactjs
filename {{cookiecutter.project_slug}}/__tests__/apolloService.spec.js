import apolloService from 'apolloService';

describe('The apollo service', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = apolloService;
  });

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
