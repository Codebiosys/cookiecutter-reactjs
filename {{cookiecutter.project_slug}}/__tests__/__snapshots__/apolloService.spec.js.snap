// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The apollo service renders 1`] = `
ApolloClient {
  "addTypename": true,
  "dataId": [Function],
  "dataIdFromObject": [Function],
  "disableNetworkFetches": false,
  "fieldWithArgs": [Function],
  "fragmentMatcher": HeuristicFragmentMatcher {},
  "initialState": Object {},
  "middleware": [Function],
  "mutate": [Function],
  "networkInterface": HTTPFetchNetworkInterface {
    "_afterwares": Array [],
    "_middlewares": Array [],
    "_opts": Object {},
    "_uri": "http://:4000/graphql",
  },
  "query": [Function],
  "queryDeduplication": true,
  "reducerConfig": Object {
    "addTypename": true,
    "customResolvers": undefined,
    "dataIdFromObject": [Function],
    "fragmentMatcher": [Function],
  },
  "resetStore": [Function],
  "setStore": [Function],
  "ssrMode": false,
  "version": "1.8.0",
  "watchQuery": [Function],
}
`;
